### ISSUE TITLE
question about fully_actuated hexarotor

### ISSUE THREAD

**Issue by lygoon** [zejdajan/sbs-sem-project-2#2]
dear Dr. Zejdajan，
    Excuse me. I have a question about the fully_actuated hexarotor and the firmware. I refer to the firmware of the fully_actuated hexarotor released by you, and after a modification, I burned it to the Pixhawk2.4.8. After calibration, when I use the RC to push up the throttle, However, the speed of the motor did not increase, and it has been idling. Therefore, I would like to ask why this has been idling and how to solve it? I would be very grateful if you could answer my question.

**Comment by zejdajan** [comment]
Hi, what about simulation? Is it working properly?

**Comment by lygoon** [comment]
Yes, there is no problem with the simulation, it can take off and land, only a little bit floating. But real hexa can't.

**Comment by zejdajan** [comment]
I am not sure bcs it is more than a year since i was working on it. However I think that once you flash PX4 firmware into board, in QGROUNDCONTROL you have to change one config variable, not sure which one exactly now.

**Comment by lygoon** [comment]
What is the configuration variable? I have modified the following parameters in QGC: CBRK_IO_SAFETY, CBRK_SUPPLY_CHK, CBRK_USB_CHK, COM_POWER_COUNT, but none of these parameters can solve my problem, that is, using the joystick in RC to increase the thrust but the motor speed does not increase.

**Comment by zejdajan** [comment]
OK, where have you connected ESC controls? IO PWM OUT or FMU PWM OUT? Maybe try the the other one.

**Comment by lygoon** [comment]
Sorry, I didn't fully understand what you meant. In terms of hardware, I connected the control wire of the ESC to the "MAIN OUT" port on the Pixhawk. There are "MAIN OUT" ports and "AUX OUT" ports on the Pixhawk. Do you mean to modify a certain parameter in QGC, about IO PWM OUT or FMU PWM OUT?

**Comment by zejdajan** [comment]
I was using power management board on my drone. I am not sure but I think that I had to change cable from low left to low rigth or otherwise (FMU vs IO PWM OUT)

**Comment by lygoon** [comment]
I don't quite understand what you said, I want to confirm whether it is a hardware problem or a code problem. The following picture is a schematic diagram of the connection of some components of my drone that I drew. When I use this connection method for ordinary six-rotor drones, there is no problem. Could you please explain it a bit more? It would be better with a picture.
![picture](https://github.com/zejdajan/sbs-sem-project-2/assets/87755127/e6ea16e8-a76b-4bb1-b6f0-cf17cff6784a)

**Comment by zejdajan** [comment]
Hi, I dont have a drone available, so I cant tell exactly, its a long time. I was following this manual with Pixhawk 4+GPS: https://ctu-mrs.github.io/docs/hardware/px4_configuration.html
Difference could me only in this: Connect a cable for the ESC signals to the IO-PWM-in port (if you soldered the ESC signal cables) or the FMU-PWM-in port (if you used the servo connectors). ---I dont know which port I used

**Comment by lygoon** [comment]
Oh, I see what you mean. You use the pixhawk 4, and use the supporting power management board. But now that pixhawk 4 is discontinued, I had to use pixhawk, pixhawk 2.4.8 and cuav v5 to try, but they all have the same problem.   
![picture2](https://github.com/zejdajan/sbs-sem-project-2/assets/87755127/25d9a9c6-855b-4a35-bdb3-6d6ec695d6fd)

**Comment by lygoon** [comment]
Dear Dr. Zejdajan, now I use a pixhawk 4. According to what you said, in the wiring, I connected the IO PWM OUT port of the pixhawk 4 to the FMU PWM IN port of the power management board. After a series of calibrations, when the motor test was performed on QGC, QGC displayed "motor test command rejected", so I adjusted the parameter "CBRK_IO_SAFETY" from the default "0" to the maximum value, and the motor immediately rotated after restarting up. In this case, I turned off the motor using the emergency kill button on the RC, but as soon as the "emergency kill button" is turned on, the motor spins itself again. At this time, test a single motor in QGC, drag the slider corresponding to a certain motor upward, and the motor can rotate normally. Then, I changed the "COM_PREARM_MODE" in QGC from the default "safety buttom" to "disabled". After restarting, the motor will not rotate automatically when the battery is connected, and the remote control can also unlock the hexa, but put the thrust lever up When pushing, the speed of the motor will not increase, and it will always be at idle speed, which cannot be controlled. This was my problem at the beginning too. Could you please tell me how to solve this one?

![2](https://github.com/zejdajan/sbs-sem-project-2/assets/87755127/460d13f7-1faf-4564-abc0-565a63d29f0c)

![3](https://github.com/zejdajan/sbs-sem-project-2/assets/87755127/d3de9c5c-b558-4947-9b2f-688648573695)

**Comment by zejdajan** [comment]
I was using Garmin GPS with arming button. What happens if you switch cable going to the Pixhawk into IOPWMOUT to FMU PWM OUT? Cable connected into the powerboard is according to the link I sent you.

**Comment by lygoon** [comment]
When I restored the two parameters modified above, and plugged in the GPS that comes with pixhawk 4, and unlocked the hexa with RC, then QGC displayed: Arming denied: Press safety switch first. When I short press the "switch" on the GPS, it can't be unlocked. Only when I press the "switch" for a long time, it will be unlocked, but at this time all the motors will start up immediately, and the RC can't control the motors, and the speed is always idling. When the "IO PWM OUT" on the pixhawk 4 is switched to "FMU PWM OUT", connect the power supply, and all the motors will immediately emit continuous and high-frequency beeping sound.

**Comment by zejdajan** [comment]
Then, I am out of ideas, sorry I could help you. I dont have a drone available anymore to test it,.

**Comment by lygoon** [comment]
There is a problem, in your firmware, there are 1111_fully_actuated and 6666_fully_actuated two fully-actuated frames under the folder "/ROMFS/px4fmu_common/init.d/airframes", which one do you choose in QGC? I want to rule out the possibility of a firmware bug. I chose the frame starting with 1111.
